lib/helper/WebDriverIO.js

```diff
@@ -560,153 +700,211 @@ class WebDriverIO extends Helper {
 
   /**
    * {{> ../webapi/grabTextFrom }}
+   * Appium: support
    */
   grabTextFrom(locator) {
-    return this.browser.getText(withStrictLocator(locator)).then(function (text) {
+    return this.browser.getText(withStrictLocator(locator)).then(function(text) {
       return text;
     });
   }
 
   /**
    * Retrieves the innerHTML from an element located by CSS or XPath and returns it to test.
    * Resumes test execution, so **should be used inside a generator with `yield`** operator.
+   * Appium: support only web testing
    *
    * ```js
    * let postHTML = yield I.grabHTMLFrom('#post');
    * ```
    */
   grabHTMLFrom(locator) {
-    return this.browser.getHTML(withStrictLocator(locator)).then(function (html) {
-      return html;
-    });
+    if (this.options.desiredCapabilities.browserName) {
+      return this.browser.getHTML(withStrictLocator(locator)).then(function(html) {
+        return html;
+      });
+    } else {
+      throw new Error(`grabHTMLFrom method can't be used in apps testing`);
+    }
   }
 
   /**
    * {{> ../webapi/grabValueFrom }}
-  */
+   * Appium: support only web testing
+   */
   grabValueFrom(locator) {
-    return this.browser.getValue(withStrictLocator(locator)).then(function (text) {
-      return text;
-    });
+    if (this.options.desiredCapabilities.browserName) {
+      return this.browser.getValue(withStrictLocator(locator)).then(function(text) {
+        return text;
+      });
+    } else {
```

Comment:
Ok, will create 3 classes: WebdriverIOBasic (for web and apps), WebdriverIO extends WebdriverIOBasic (for web only) and Appium extends WebdriverIOBasic (for apps only)